test/streaming-client.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { WebSocketClient, WebSocketConnection } from 'websocket';
+import { AudioConfig } from '../src/models/streaming/AudioConfig';
+import { BufferedDuplex } from '../src/models/streaming/BufferedDuplex';
+import { RevAiStreamingClient } from '../src/streaming-client';
+const fs = require('fs');
+const events = require('events');
+
+jest.useFakeTimers();
+
+let sut: RevAiStreamingClient;
+let mockClient: WebSocketClient;
+
+const audioConfig = new AudioConfig("audio/x-wav");
+const token = "testToken";
+
+describe('streaming-client', () => {
+    beforeEach(() => {
+        sut = new RevAiStreamingClient(token, audioConfig);
+        mockClient = sut.client;
+    });
+
+    describe('start', () => {
+        it('Connects to api with parameters', () => {
+            // Act
+            const res = sut.start();
+
+            // Assert
+            expect(mockClient.connect).toBeCalledWith(`wss://api.rev.ai/speechtotext/v1alpha/stream?` +
+                `access_token=${token}` +
+                `&content_type=${audioConfig.getContentTypeString()}`
+            );
+            expect(mockClient.connect).toBeCalledTimes(1);
+        });
+
+        it('Returns duplex stream', () => {
+            // Act
+            const res = sut.start();
+
+            // Assert
+            expect(res).toBeInstanceOf(BufferedDuplex);
+        });
+
+        it('adds event listener to httpResponse', () => {
+            // Setup
+            const res = sut.start();
+            let statusCode = null;
+            sut.on('httpResponse', code => {
+                statusCode = code;
+            });
+
+            // Act
+            mockClient.emit('httpResponse', {statusCode: 401});
+
+            // Assert
+            expect(statusCode).toBe(401);
```

Comment:
should we also assert that .end() is called?

Reply:
:+1: